Update definition on idempotency for Activities (#2208)
temporalio/documentation: docs-src/concepts/what-is-an-activity-definition.md

Temporal recommends that Activities be idempotent.

> Idempotent means that performing an operation multiple times will have the same result as performing it once. In the context of Temporal, Activities should be designed in a way that they can be safely executed multiple times without causing unexpected or undesired side effects.

Reviewer: ```suggestion
Idempotent means that performing an operation multiple times has the same result as performing it once.
In the context of Temporal, Activities should be designed to be safely executed multiple times without causing unexpected or undesired side effects.
```

Temporal recommends that Activities be idempotent.

Idempotent means that performing an operation multiple times has the same result as performing it once.

> In the context of Temporal, Activities should be designed to be safely executed multiple times without causing unexpected or undesired side effects.

Reviewer 2: ```suggestion
In the context of Temporal, Activities should usually be designed to be safely executed multiple times without causing unexpected or undesired side effects.
```

don't need to if activity is non-local an max attempts = 1

Reviewer 3: Non local activities with max attempts = 1 may still be executed more than once. 

EX:
* Worker gets activity task, starts execution, and then loses connection with server. The activity starts executing
* Server times out the task and retries
* Starts again on another worker

Exactly-once semantics are impossible. For a lot of people a lot of the time, that's fine, but highly sensitive applications like finance _must_ ensure they protect against multiple invocations.

Reviewer 2: > Worker gets activity task

This atomically counts as an attempt, right? So Server wouldn't retry after timeout?

I know we can't do exactly once, but thought I remembered from roey or max that we can do at most once and this is how.

Reviewer 3: Ah, yup fair point. That would indeed avoid it in this case - of course most people are probably going to want retries, but yeah I think your original point is valid. Ignore me :)

Reviewer 2: when I read/hear it, I hear always, so "should usually" to me means there are times when it should not be / doesn't have to be
